design/m1_push_to_file_design.md

```diff
@@ -0,0 +1,1247 @@
+annotation) with contents that look like the following:
+
+```
+url: https://database.example.com
+port: 5
+password: my-secret-p@$$w0rd
+username: postgres
+```
+
+###### Example `conjur-secrets` List With Aliases and `conjur-secrets-path` Prefix
+
+The following is an example of a `conjur.org/conjur-secrets.{secret-group}`
+Annotation that defines a list of Conjur variable paths, including some that
+use an alias, for a hypothetical secrets group named `cache`:
+
+```
+    conjur.org/conjur-secrets.cache: |
+      - url
+      - admin-password: password
+      - admin-username: username
+
+    conjur.org/conjur-secrets-path.cache: dev/memcached/
+
+```
+
+In this example, the paths for the secrets that are retrieved from Conjur are:
+
+- dev/memcached/url
+- dev/memcached/password
+- dev/memcached/username
+
+and the names of the secrets that are written to the secrets file are:
+
+- url
+- admin-password
+- admin-username
+
+so that the `cache.yaml` file that Secrets Provider writes includes:
+
+```
+url: https://database.example.com
+admin-password: my-secret-p@$$w0rd
+admin-username: admin
+```
+
+###### Example `conjur.org/secret-file-template.{secret-group}`
+
+The following is an example of a custom secrets file template that uses the
+Golang [`text/template`](https://golang.org/pkg/text/template/) syntax.
+Variables can be referenced via their name
+(their alias if provided, otherwise, the last word in their Conjur variable
+path) and are replaced with their value from Conjur. (Note that the
+`conjur.org/secret-file-path.{secret-group}` annotation is required when
+a custom secrets file template is configured):
+
+```
+    conjur.org/secret-file-path.cache: /conjur/secrets/cache.cfg
+    conjur.org/secret-file-template.cache: |
+      {
+        "cache": {
+          "url": {{ .url }},
+          "password": {{ .admin-password }},
+          "username": {{ .admin-username }},
+          "port": 123456
+        }
+      }
+```
+
+The Secrets Provider will write a file `/conjur/secrets/cache.cfg`
+with contents that look like the following:
+
+```
+{
+  "cache": {
+    "url": cache.example.com,
+    "password": my-secret-p@$$w0rd,
+    "username": admin,
+    "port": 123456
+  }
+}
+```
+
+#### Example Kubernetes Manifests and Templates
+
+For reference,
+[Kubernetes manifests and Helm named templates](#example-kubernetes-manifests-and-helm-named-templates)
+for an example application that uses a Secrets Provider init container in
+Push-to-File mode are listed in the Appendix.
+
+#### Output File Formats
+
+##### Example YAML Secrets File
+
+```
+url: https://database.example.com
+admin-password: p@$$w0rd
+admin-username: zappa
+```
+
+##### Example JSON Secrets File
+
+```
+{
+   "url": "https://database.example.com",
+   "admin-password": "p@$$w0rd",
+   "admin-username": "zappa"
+}
+```
+
+##### Example `bash` Secrets File
+
+```
+   export URL="https://database.example.com"
+   export ADMIN_PASSWORD="p@$$w0rd"
```

Comment:
Oh, I like @andytinkham idea about validating the resulting file to avoid any issues.  Doesn't feel required for MVP  though.  Maybe just take note of it, @diverdane?